File: CS502024/sentimental-readability/readability.py
import cs50


def main():
    text = cs50.get_string("Text: ")
    letters = 0
    words = 0
    sent = 0
    for i in range(0, len(text)):
        if text[i].isalpha():
            letters += 1
        elif text[i] == ' ':
            words += 1
        elif text[i] == '.' or text[i] == '!' or text[i] == '?':
            sent += 1
    words += 1
    cli = 0.0588 * average_by_100(letters, words) - 0.296 * average_by_100(sent, words) - 15.8
    grade = round(cli)
    if cli < 1:
        print("Before Grade 1")
    elif (cli > 16):
        print("Grade 16+\n")
    else:
        print("Grade", grade)


def average_by_100(a, b):
    return 100 * a / b


main()

File: CS502024/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>

float averageBy100(int a, int b);

int main(void)
{
    string text = get_string("Text: ");
    int letters = 0;
    int words = 0;
    int sent = 0;
    float CLI = 0;
    int grade;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sent++;
        }
    }
    words++;
    CLI = 0.0588 * averageBy100(letters, words) - 0.296 * averageBy100(sent, words) - 15.8;
    grade = round(CLI);
    if (CLI < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (CLI > 16)
    {
        printf("Grade 16+\n");
    }
    else
        printf("Grade %i\n", grade);
    return 0;
}

float averageBy100(int a, int b)
{
    return 100.0 * a / b;
}
